prober/utils_test.go

```diff
@@ -147,22 +147,40 @@ func TestChooseProtocol(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping network dependent test")
 	}
-	ctx := context.Background()
-	registry := prometheus.NewPedanticRegistry()
-	w := log.NewSyncWriter(os.Stderr)
-	logger := log.NewLogfmtLogger(w)
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	checkChosenProtocol(context.Background(), false, logger, t)
+	checkChosenProtocol(context.Background(), true, logger, t)
+}
 
-	ip, _, err := chooseProtocol(ctx, "ip4", true, "ipv6.google.com", registry, logger)
-	if err != nil {
-		t.Error(err)
-	}
+func checkChosenProtocol(ctx context.Context, randomResolvedIP bool, logger log.Logger, t *testing.T) {
+	checkChosenProtocolWithoutFallback(ctx, "ip4", randomResolvedIP, "ipv6.google.com", logger, t)
```

Comment:
I'd like to use something like https://sslip.io/ to validate exact IP returned by `chooseProtocol` (when `randomResolvedIP` is `false`), but sslip.io doesn't support multiple IPs, which limits such tests to a single (IPv4 or IPv6) address and makes them less helpful (though testing the case with single IP address is still useful, because there are branches for this case).